.github/workflows/scripts/check_bundle_size.js

```diff
@@ -0,0 +1,67 @@
+module.exports = {
+  checkBaseBundleSize: () => {
+    const fs = require('fs');
+    const jsonString = fs.readFileSync('.github/analyze/bundles.json');
+    const data = JSON.parse(jsonString);
+    const pagesToCheck = [
+      '/',
+      '/cli/start/install',
+      '/lib/auth/getting-started/q/platform/[platform]',
+      '/start',
+      '/cli'
+    ];
+    const baseBundleSizes = [];
+    data.pages.filter((page) => {
+      if (pagesToCheck.includes(page.label)) {
+        baseBundleSizes.push({
+          page: page.label,
+          gzipSize: page.gzipSize,
+          parsedSize: page.parsedSize,
+          statSize: page.statSize,
+          totalGzipSize: page.totalGzipSize,
+          totalParsedSize: page.totalParsedSize,
+          totalStatSize: page.totalStatSize
+        });
+      }
+    });
+    return baseBundleSizes;
+  },
+
+  checkHeadBundleSize: () => {
+    const fs = require('fs');
+    const jsonString = fs.readFileSync('.github/analyze/bundles.json');
+    const data = JSON.parse(jsonString);
+    const pagesToCheck = [
+      '/',
+      '/cli/start/install',
+      '/lib/auth/getting-started/q/platform/[platform]',
+      '/start',
+      '/cli'
+    ];
+    const headBundleSizes = [];
+    data.pages.filter((page) => {
+      if (pagesToCheck.includes(page.label)) {
+        headBundleSizes.push({
+          page: page.label,
+          gzipSize: page.gzipSize,
+          parsedSize: page.parsedSize,
+          statSize: page.statSize,
+          totalGzipSize: page.totalGzipSize,
+          totalParsedSize: page.totalParsedSize,
+          totalStatSize: page.totalStatSize
+        });
+      }
+    });
+    return headBundleSizes;
+  },
+
+  compareBundles: (baseBundles, headBundles) => {
+    for (let i = 0; i < baseBundles.length; i++) {
+      if (baseBundles[i] !== headBundles[i]) {
```

Comment:
Is the purpose of this `if` statement just to skip over the comparison if the same object is passed into the function as `baseBundles` and `headBundles`?  Are we guaranteed that the pages Array generated will always be in the same order or is there a situation where the filter function from the `checkBundleSize` function produces arrays that are in a different order between head and base (such as pages being added or removed)?

Reply:
Fixed this logic to check more explicitly.